### PR TITLE
functional-tester: defrag every 500 round

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/4665.

### CODE REVIEW

**Comment by xiang90** [comment]
i would love to see current testing become stable, then to merge this in.

**Comment by gyuho** [comment]
@xiang90 Agree. We need to get the compaction, hash fixed first and run them for awhile.

Will revisit when those are stable.

**Comment by xiang90** [comment]
Probably every 500 rounds will be good enough (1round/minute -> 500 minutes to defrag once)

**Comment by gyuho** [comment]
@xiang90 Just fixed. PTAL. Thanks!

File: tools/functional-tester/etcd-tester/cluster.go [hunk 1]
```diff
@@ -376,3 +376,20 @@ func (c *cluster) checkCompact(rev int64) error {
...
+		}
+		mt := pb.NewMaintenanceClient(conn)
+		if _, err = mt.Defragment(context.Background(), &pb.DefragmentRequest{}); err != nil {
+			return err
+		}
+		conn.Close()
+		plog.Printf("Defragmented %s", u)
+	}
+	return nil
+}
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/cluster.go]: \n at the end of the file?

File: tools/functional-tester/etcd-tester/cluster.go [hunk 2]
```diff
@@ -376,3 +376,20 @@ func (c *cluster) checkCompact(rev int64) error {
 	}
 	return nil
 }
+
+func (c *cluster) defrag() error {
+	for _, u := range c.GRPCURLs {
+		plog.Printf("Defragmenting %s", u)
+		conn, err := grpc.Dial(u, grpc.WithInsecure(), grpc.WithTimeout(5*time.Second))
+		if err != nil {
+			continue
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/cluster.go]: return err? why continue here?

File: tools/functional-tester/etcd-tester/cluster.go [hunk 3]
```diff
@@ -376,3 +376,20 @@ func (c *cluster) checkCompact(rev int64) error {
 	}
 	return nil
 }
+
+func (c *cluster) defrag() error {
+	for _, u := range c.GRPCURLs {
+		plog.Printf("Defragmenting %s", u)
```
Comment (xiang90) [inline tools/functional-tester/etcd-tester/cluster.go]: defragmenting? (to be consistent with line 169 in tester.go)

**Comment by xiang90** [comment]
LGTM

**Comment by gyuho** [comment]
Just fixed up (`defrag...` to be consistent, make it return err). Thanks.

**Comment by gyuho** [comment]
Will redeploy testing cluster with this patch.

**Pull request merged**